src/tests/utils/common.cairo

```diff
@@ -36,6 +37,21 @@ pub fn declare_and_deploy(contract_name: ByteArray, calldata: Array<felt252>) ->
     deploy(contract_class, calldata)
 }
 
+pub fn declare_and_deploy_with_caller(
+    contract_name: ByteArray, calldata: Array<felt252>, caller: ContractAddress
+) -> ContractAddress {
+    let contract_class = declare(contract_name).unwrap_syscall();
+
+    // Precalculate the address before the constructor call (deploy) itself
+    let contract_address = contract_class.precalculate_address(@calldata);
+
+    start_cheat_caller_address(contract_address, caller);
+    deploy(contract_class, calldata);
+    stop_cheat_caller_address(contract_address);
+
+    contract_address
+}
```

Comment:
Not sure I get why this is needed. We don't initialize the state with `get_caller_address` because it's an anti-pattern, no? Or am I missing something?